### PR TITLE
[MISC] Support python 3.8, 3.9, 3.10

### PR DESCRIPTION
**Pull request by NicolaDonelli**
Closes issue #1, also adding compatibility checks with python 3.8 and 3.9

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -110,11 +112,14 @@ lint: setup_ci
...
 
-clean: uninstall
-	rm -rf docs
-	rm -rf build
+clean:
 	rm -rf $(shell find . -name "*.pyc") $(shell find . -name "__pycache__")
-	rm -rf dist *.egg-info .mypy_cache .pytest_cache .make_cache $(env_tag) $(env_ci_tag) $(install_tag)
+	rm -rf *.egg-info .mypy_cache .pytest_cache .make_cache $(env_tag) $(env_ci_tag) $(install_tag)
+
+#clean: uninstall clean_cache
```
Comment (deusebio) [inline Makefile:122]: can we remove this?

File: Makefile [hunk 2]
```diff
@@ -98,10 +100,10 @@ uninstall:
```
Comment (deusebio) [inline Makefile:100]: You forgot to add the "--no-deps" to pip

**Review by deusebio — CHANGES_REQUESTED** [review]
Just a couple of comments. 

I only have an issue with the commit message. The commit is 2% about providing support to python 3.8, 3.9 and 3.10 (you only udated requirements and CI pipeline, right?) and the majority of modifications are about documentation. 

If possible, I belive it would be better to separate the two things in two different commits

**Comment by NicolaDonelli** [comment]
Yup, you are right. I splitted the commit in two added the `--no-deps` flag to pip install tar and removed the commented lines.

**Review by deusebio — APPROVED** [review]

**Pull request merged**